target/linux/ramips/dts/mt7621_zyxel_lte3301-plus.dts

```diff
@@ -0,0 +1,217 @@
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "lte3301-plus:white:power";
+			gpios = <&gpio 5 GPIO_ACTIVE_HIGH>;
+		};
+
+		wifi {
+			label = "lte3301-plus:white:wifi";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		internet {
+			label = "lte3301-plus:white:internet";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
+
+		usb {
+			label = "lte3301-plus:white:usb";
+			gpios = <&gpio 24 GPIO_ACTIVE_LOW>;
+		};
+
+		lte {
+			label = "lte3301-plus:white:lte";
+			gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
+		};
+
+		mobile_green {
+			label = "lte3301-plus:green:mobile";
+			gpios = <&gpio 31 GPIO_ACTIVE_LOW>;
+		};
+
+		mobile_orange {
+			label = "lte3301-plus:orange:mobile";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+		};
+
+		mobile_red {
+			label = "lte3301-plus:red:mobile";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+	};
+
+};
+
+&gpio {
+	status = "okay";
+
+	enable_usb_power {
+		gpio-hog;
+		line-name = "enable USB power";
+		gpios = <7 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "uboot";
+			reg = <0x00000 0x80000>; /* 64 KB */
+		};
+		partition@80000 {
+			label = "uboot_env";
+			reg = <0x80000 0x80000>; /* 64 KB */
+		};
+		factory: partition@100000 {
+			label = "factory";
+			reg = <0x100000 0x40000>;
+		};
+		partition@140000 {
+			label = "Kernel";
+			reg = <0x140000 0x1ec0000>;
+		};
+		partition@2140000 {
+			label = "kernel2";
+			reg = <0x2140000 0x1ec0000>;
+		};
+		partition@4000000 {
+			label = "wwan";
+			reg = <0x4000000 0x100000>;
+		};
+		partition@4100000 {
+			label = "ubi";
+			reg = <0x4100000 0x3ee0000>;
+		};
+	};
+};
+
+&ethernet {
+	pinctrl-0 = <&mdio_pins>, <&rgmii1_pins>;
+};
+
+&state_default {
+	gpio {
+		groups = "i2c", "rgmii2", "uart3", "jtag", "wdt";
+		function = "gpio";
+	};
+
+};
+
+&gmac0 {
+	nvmem-cells = <&macaddr_factory_fe6e>;
+	nvmem-cell-names = "mac-address";
+};
+
+&switch0 {
+	ports {
+		port@0 {
+			status = "okay";
+			label = "lan4";
+		};
+		port@1 {
+			status = "okay";
+			label = "lan3";
+		};
+		port@2 {
+			status = "okay";
+			label = "lan2";
+		};
+		port@3 {
+			status = "okay";
+			label = "lan1";
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	status = "okay";
+	mt7615d@0,0 {
+		/* In reality  at hangs at pcie1, this is a driver bug */
+		compatible = "pci14c3,7615";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,firmware-eeprom = "mt7615e_eeprom.bin";
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+		nvmem-cells = <&macaddr_factory_fe6e>;
+		nvmem-cell-names = "mac-address";
+		mac-address-increment = <(1)>;
+	};
+
+};
+
+&factory {
+	compatible = "nvmem-cells";
+	#address-cells = <1>;
+	#size-cells = <1>;
+	mtd-mac-address = <&factory 0xfe6e>;
```

Comment:
Why this?

Reply:
I don't understand? It's for the device to get it's proper MAC. Something wrong?

Reply:
this is an old binding... we now use nvmem cells